packages/opencensus-core/test/test-metric-registry.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {LabelKey} from '../src/metrics/export/types';
+import {MetricRegistry} from '../src/metrics/metric-registry';
+import {MeasureUnit} from '../src/stats/types';
+
+const METRIC_NAME = 'metric-name';
+const METRIC_DESCRIPTION = 'metric-description';
+const UNIT = MeasureUnit.UNIT;
+const LABEL_KEYS: LabelKey[] = [{key: 'code', description: 'desc'}];
+const LABEL_KEYS_WITH_NULL: LabelKey[] =
+    [{key: 'code', description: 'desc'}, null];
+
+const registry = new MetricRegistry();
+
+describe('addInt64Gauge', () => {
+  it('should throw an error when the name is null', () => {
+    assert.throws(() => {
```

Comment:
Could you please add coverage for empty string, 0 and undefined?
The same for other tests in this PR

Reply:
Added tests for empty string and undefined values.

Reply:
What about `0` and `'0'`?

Reply:
Typescript will enforce type checking, preventing "0" (unless someone uses an `any` forced type), but if someone is using this class in pure javascript you can't enforce types and could receive any input.

I guess the question is really if this is an external facing interface or an internal only interface where you can expect that the inputs have already been sanitized and validated.